src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -158,9 +207,34 @@ def save(self, *args, **kwargs):
         """
         if not self.slug:
             self.slug = self.generate_slug()
-
         super().save(*args, **kwargs)
 
+    def set_ancestors_are_public(self):
+        # Update ancestors_are_public for the sub tree.
+        # note: This doesn't trigger a new save for each instance, rather
+        # just a SQL update.
+        included_in_non_public_subtree = self.__class__.objects.filter(
+            is_public=False, path__rstartswith=OuterRef("path"), depth__lt=OuterRef("depth")
+        )
+        self.get_descendants_and_self().update(
```

Comment:
@jaroel is it possible to change this query which is mysql compatible as well.

> MySQL doesn't allow updating the table you are already using in an inner select as the update criteria.

Reply:
Dunno, I don't use mysql. What did you already try but didn't work?